pylint/checkers/refactoring.py

```diff
@@ -261,13 +266,42 @@ def _check_redefined_argument_from_local(self, name_node):
                                  args=(name_node.name, ))
 
     @utils.check_messages('redefined-argument-from-local',
-                          'too-many-nested-blocks')
+                          'too-many-nested-blocks',
+                          'mutable-sequence-modified-in-loop')
     def visit_for(self, node):
         self._check_nested_blocks(node)
 
         for name in node.target.nodes_of_class(astroid.AssignName):
             self._check_redefined_argument_from_local(name)
 
+        if self._check_iter_is_infer_list(node):
+            for part in node.nodes_of_class(astroid.Expr):
```

Comment:
We just need to find calls, not exprs. Also, when using ``nodes_of_class`` make sure the object that is yielded has the same scope as the scope of the node you want to iterate through, since ``nodes_of_class`` can return values from inner scopes. We don't probably have a lot of these cases, but should be handled nevertheless.